src/elements/ExactCFbend.H

```diff
@@ -0,0 +1,665 @@
+                   {
+                       amrex::ParticleReal const pzf = std::sqrt(powi<2>(pperp)-powi<2>(pxout));
+                       amrex::ParticleReal const theta = slice_phi + std::asin(px/pperp) - std::asin(pxout/pperp);
+
+                       // update position coordinates
+                       xout = -m_rc + rho*cos_phi + m_rc*(pzf + px*sin_phi - pzi*cos_phi);
+                       yout = y + theta * m_rc * py;
+                       tout = t - theta * m_rc * (pt - 1.0_prt * m_ibeta) - slice_phi * m_rc * m_ibeta;
+
+                   } // else { amrex::Print() << "Warning: outside map domain." << "\n"; }
+               } // else { amrex::Print() << "Warning: outside map domain." << "\n"; }
+            }
+
+            // push particle coordinates
+            particle(1) = xout;
+            particle(2) = pxout;
+            particle(3) = yout;
+            particle(4) = pyout;
+            particle(5) = tout;
+            particle(6) = ptout;
+
+            zeval += 0_prt;
+        }
+
+        /** This pushes a particle in an ExactMultipole element through
+         *  the symplectic map associated with H_2 in the Hamiltonian
+         *  splitting H = H_1 + H_2.  This is the exact kick obtained
+         *  in the ideal multipole field.
+         *
+         * @param tau Map step size in m
+         * @param[in,out] particle particle phase space 6-vector
+         * @param[in,out] zeval Longitudinal on-axis location in m
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        void map2 (amrex::ParticleReal const tau,
+                   amrex::SmallVector<amrex::ParticleReal, 6, 1> & particle,
+                   amrex::ParticleReal & zeval) const
+        {
+            using namespace amrex::literals; // for _rt and _prt
+            using namespace amrex::Math;  // for powi
+
+            amrex::ParticleReal const x = particle(1);
+            amrex::ParticleReal const px = particle(2);
+            amrex::ParticleReal const y = particle(3);
+            amrex::ParticleReal const py = particle(4);
+            amrex::ParticleReal const t = particle(5);
+            amrex::ParticleReal const pt = particle(6);
+
+            amrex::ParticleReal xout = x;
+            amrex::ParticleReal pxout = px;
+            amrex::ParticleReal yout = y;
+            amrex::ParticleReal pyout = py;
+            amrex::ParticleReal tout = t;
+            amrex::ParticleReal ptout = pt;
+
+#if AMREX_DEVICE_COMPILE
+            amrex::ParticleReal* k_normal = m_k_normal_d_data;
+            amrex::ParticleReal* k_skew = m_k_skew_d_data;
+#else
+            amrex::ParticleReal* k_normal = m_k_normal_h_data;
+            amrex::ParticleReal* k_skew = m_k_skew_h_data;
+#endif
+
+            // defined dimensionless variables scaled by radius of curvature
+            amrex::ParticleReal rho = 1_prt + x / m_rc;
+            amrex::ParticleReal y_scl = y / m_rc;
+            amrex::ParticleReal ln_rho = log(rho);
```

Comment:
Single precision support:
```suggestion
            amrex::ParticleReal ln_rho = std::log(rho);
```